x_transformers/x_transformers.py

```diff
@@ -127,10 +127,12 @@ def __init__(self, dim, max_seq_len, l2norm_embed = False):
         self.l2norm_embed = l2norm_embed
         self.emb = nn.Embedding(max_seq_len, dim)
 
-    def forward(self, x):
-        n = torch.arange(x.shape[1], device = x.device)
-        pos_emb = self.emb(n)
-        pos_emb = rearrange(pos_emb, 'n d -> () n d')
+    def forward(self, x, pos = None):
+        if not exists(pos):
+            pos = torch.arange(x.shape[1], device = x.device)
+        pos_emb = self.emb(pos)
+        if pos_emb.ndim == 2:
```

Comment:
i actually learned that when doing operations on two tensors with unequal number of dimensions, the one(s) with less dimensions are unsqueezed from the left (tensors are right aligned before op)

so i think you can safely remove this conditional!

Reply:
Makes sense, you mean removing the `rearrange` altogether, right?

Reply:
yup, was about to do it, but you are in a good position to just remove it altogether :pray: